rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -3157,12 +3157,9 @@ private Federation getFederationFromTxOutputs(BtcTransaction btcTx) {
         }
 
         // and then against retiring fed
-        Federation retiringFederation = getRetiringFederation();
-        if (retiringFederation != null && outputsMatchFederation(btcTx, retiringFederation)) {
-            return retiringFederation;
-        }
-
-        throw new IllegalStateException("Couldn't extract federation from btcTx outputs.");
+        return getRetiringFederation()
+            .filter(retiringFed -> outputsMatchFederation(btcTx, retiringFed))
+            .orElseThrow(() -> new IllegalStateException("Couldn't extract federation from btcTx outputs."));
```

Comment:
```suggestion
        if (outputsMatchFederation(btcTx, retiringFederation) {
            return retiringFederation;
        }
        
        throw new IllegalStateException("Couldn't extract federation from btcTx outputs.");
```
you already assumed you have the retiring fed right?

Reply:
also i'll modify the message error to say "Couldn't extract _retiring_ federation from btcTx outputs."
(no big deal at all)